CounterApp/App/AppView.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  AppView.swift
+//  CounterApp
+//
+//  Created by Wenjuan Li on 2022/7/7.
+//
+
+import ComposableArchitecture
+import StoreKit
+import SwiftUI
+
+struct AppView: View {
+    let store: Store<AppState, AppAction>
+
+    var body: some View {
+        WithViewStore(store.scope(state: \.viewState)) { viewStore in
+            NavigationView {
+                VStack {
+                    Text("\(viewStore.counter)")
+
+                    NavigationLink(
+                        "Goto edit page",
+                        destination: CounterView(
+                            store: self.store.scope(
+                                state: { appState in appState.counter },
+                                action: { localAction in AppAction.counter(localAction) }
+                            )
+                        )
+                    )
+                    .padding()
+
+                    Button("Show Lock View") {
+                        viewStore.send(.lock(LockAction.setSheet(isPresented: true)))
+                    }
+                }
+                .font(Font.title2)
+            }
+            .sheet(
+                isPresented: viewStore.binding(
+                    get: \.isShowLockView,
+                    send: { localState in
+                        AppAction.lock(LockAction.setSheet(isPresented: localState))
```

Comment:
Just to mention that in most of the cases where these closures can be satisfied by either key paths or enum cases, we still want that syntax. This was mentioned in the videos just to highlight the fact that the argument here is much more flexible to allow for more complex computations of bindings. Same applies to Store's scope method.